### DISCUSSION TITLE
[indexer] Get ready for nearcore 1.23.0

### DISCUSSION THREAD

**Discussion by khorolets** [near/nearcore#5535] [Node Public Interfaces]
`nearcore` is planned to be updated to the version of 1.23.0:

* `testnet` on Nov 29, 2021
* `mainnet` on Dec 13, 2021

Protocol upgrade is expected, also this update includes database migration which takes ~20 min on the regular node and ~5 hour on the archival one

### Quick info
---
**Commit hash** [`397a7a45b8efd3e4604374a4efabd567c03c0a19 `](https://github.com/near/nearcore/commit/397a7a45b8efd3e4604374a4efabd567c03c0a19) 1.23.0-rc.1
**Expected issues**: Indexer won't be working and available during the data migration (if you have more than one node, consider updating them one by one)
**Reference**: have a look at how [we made it on NEAR Indexer for Explorer](https://github.com/near/near-indexer-for-explorer/pull/200)

Feel free to leave your questions in the comments below.

**Comment by ghost** [comment]
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: InvalidData, error: "Not all bytes read" }', /home/admin/.cargo/git/checkouts/nearcore-5bf7818cf2261fd0/397a7a4/core/store/src/migrations.rs:788:88
```

can't upgrade

**Reply by khorolets** [reply]
I've updated the commit hash recently, can you ensure you're using the latest one https://github.com/near/nearcore/commit/397a7a45b8efd3e4604374a4efabd567c03c0a19 ?

**Reply by ghost** [reply]
thats the hash that doesn't work. i was already aware the other hash was deleted https://github.com/near/near-indexer-for-explorer/commit/0c5e9ccfed8aa9f9a0b343e9aaa82f5295f52540

**Reply by khorolets** [reply]
Can you provide more logs, what was the indexer doing before this message appeared?

**Reply by ghost** [reply]
```sh
2021-11-30T15:53:06.973913Z  INFO indexer: Load config from /home/admin/.near/testnet2/...     
2021-11-30T15:55:01.591235Z  INFO near: Opening store database at "/home/admin/.near/testnet2/data"     
2021-11-30T15:55:01.604904Z  INFO near: Migrate DB from version 29 to 30     
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: InvalidData, error: "Not all bytes read" }', /home/admin/.cargo/git/checkouts/nearcore-5bf7818cf2261fd0/397a7a4/core/store/src/migrations.rs:788:88
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Reply by khorolets** [reply]
@frol @bowenwang1996  seems like data became corrupted during the migration

**Comment by bowenwang1996** [comment]
> Protocol upgrade is expected, also this update includes database migration which takes ~20 min on the regular node and ~1 hour on the archival one

I think it definitely takes more than one hour for an archival nodes @khorolets. Where did you get the 1 hour information from?

**Reply by khorolets** [reply]
I've seen it somewhere in chats, but I've corrected to 5 hours which we've experienced on testnet

**Comment by Ljzn** [comment]
Hi, is this a MUST upgrade? We're running an old version RPC node. If the old node can keep working we don't want to bother our users.

**Reply by ghost** [reply]
yes or your indexer will stop working
